src/Microsoft.Azure.Relay/RelayedHttpListenerResponse.cs

```diff
@@ -149,11 +154,51 @@ void Dispose(bool disposing)
             }
         }
 
-        void CheckDisposed()
+        internal void SetReadOnly()
+        {
+            this.readOnly = true;
+        }
+
+        void CheckDisposedOrReadOnly()
         {
             if (this.disposed)
             {
-                throw RelayEventSource.Log.ThrowingException(new ObjectDisposedException(this.GetType().FullName), this);
+                string message = this.Context.TrackingContext.EnsureTrackableMessage(this.ToString());
```

Comment:
>this.ToString() [](start = 85, length = 15)

Did you mean this.TrackingContext.ToString()?

Reply:
Not exactly. It's customary to pass the type name in as the parameter to ObjectDisposedException (that's what the code had been doing previously). I'm keeping that plus tracking Id and address.

If I just used TrackingContext.ToString() then the ObjectDisposedException wouldn't contain the text `RelayedHttpListenerResponse` which I want to be in there.

---
In reply to: [276868207](https://github.com/Azure/azure-relay-dotnet/pull/105#discussion_r276868207) [](ancestors = 276868207)